client/components/date-range/test/index.js

```diff
@@ -155,8 +155,8 @@ describe( 'DateRange', () => {
 			const expectedStartDate = '2018-04-01';
 			const expectedEndDate = '2018-04-29';
 
-			const newStartDate = moment.utc( expectedStartDate );
-			const newEndDate = moment.utc( expectedEndDate );
+			const newStartDate = moment( expectedStartDate ).utcOffset( 0 );
+			const newEndDate = moment( expectedEndDate ).utcOffset( 0 );
```

Comment:
The tests start passing also when using a simple `moment( expectedDate )`, without setting the `utcOffset`.

Internally, `DateRange` uses the `moment()` constructor, too. That means the datetime objects are in local timezone. As we're ever only interested in the date, never in the time, it doesn't matter. The timezone offsets cancel out in the `isBefore`/`isAfter` calcuations.